File: server/tutor-domain/src/main/java/cn/forever24/tutor/profile/CorrectionStyle.java
package cn.forever24.tutor.profile;

public enum CorrectionStyle {
    LIGHT,
    STANDARD,
    STRICT;

    public static CorrectionStyle fromContractValue(String value) {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("correction style is required");
        }
        try {
            return CorrectionStyle.valueOf(value);
        } catch (IllegalArgumentException exception) {
            throw new IllegalArgumentException("unsupported correction style: " + value, exception);
        }
    }
}

File: android/app/src/main/java/cn/forever24/tutor/android/ui/CorrectionStyle.kt
package cn.forever24.tutor.android.ui

enum class CorrectionStyle(
    val label: String,
    val description: String,
) {
    LIGHT(
        label = "Light",
        description = "Keep conversation moving and save most notes for later.",
    ),
    STANDARD(
        label = "Standard",
        description = "Balance fluent practice with a few clear corrections.",
    ),
    STRICT(
        label = "Strict",
        description = "Use tighter feedback during focused practice.",
    ),
}
